### PR TITLE
Add gRPC server interceptor with golden metrics to all gRPC services [PLEN-10]

### PR DESCRIPTION
**Pull request by nicu-da**

### CODE REVIEW

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala [hunk 1]
```diff
@@ -31,12 +31,14 @@ trait OpenTelemetryFactory extends Factory {
       name: MetricName
   )(implicit
       context: MetricsContext = MetricsContext.Empty
-  ): MetricHandle.Timer =
+  ): MetricHandle.Timer = {
+    val nameWithSuffix = name :+ "duration" :+ "ms"
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:35]: Nit: In order to comply with the [Prometheus naming rules](https://prometheus.io/docs/practices/naming/#metric-names) (even if this is not `seconds` yet), wouldn't we want to suffix this with `millis` or `milliseconds` instead?

File: ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala [hunk 2]
```diff
@@ -20,7 +25,26 @@ case class OpenTelemetryMeterOwner(enabled: Boolean, reporter: Option[MetricsRep
...
+          .builder()
+          .setType(InstrumentType.HISTOGRAM)
+          .setName(new Predicate[String] {
+            override def test(t: String): Boolean = t.endsWith("duration.ms")
+          })
+          .build(),
+        View
+          .builder()
+          .setAggregation(
+            Aggregation.explicitBucketHistogram(
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala:41]: Can you possibly share the reference from where this bucket definition comes, please?

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala (hunk 1 above)
Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:35]: we plan on changing to seconds so it's irrelevant for the milliseconds

File: ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala:41]: The buckets are defined based on the data/granularity that we expect to encounter therefore there is no reference

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala (hunk 1 above)
Comment (miklos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:33]: OK, as long as we'll be switching to a standard metric name it's fine.

File: ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala (hunk 2 above)
Comment (miklos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala]: We have default buckets defined e.g. for the [Python](https://github.com/prometheus/client_python/blob/master/prometheus_client/metrics.py#L544) and [Go](https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L265) Prometheus clients that we could leverage.
I don't have a strong preference (i.e., we may keep the current bucketing). However, from a practical perspective I'm not sure if it makes sense to capture different buckets above 10s latency (i.e., do we have calls that we expect to take that long?).

**Review by miklos-da — APPROVED** [review]

File: ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala]: That's a good point, it was mostly for certain scenarios where that could be useful. But I'll default to the go defined ones for now, it makes more sense...and we can change it in the future.

Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala]: Will cover that in a separate PR

**Pull request merged**